File: solutions/julia/nth-prime/nth-prime.jl
function is_prime(n::Int)

    if n == 0 || n == 1
        return false
    end

    for m in 2:Int(ceil(sqrt(n)))
        if n != m && n % m == 0
            return false
        end
    end
    return true
end

function next_prime(n::Int)
    for m in n + 1:typemax(Int)
        if is_prime(m)
            return m
        end
    end
    throw(DomainError(n))
end

function prime(num)

    if num < 1
        throw(ErrorException("No such prime!"))
    end

    p = 0

    for i in 1:num
        p = next_prime(p)
    end

    return p

end

File: solutions/python/nth-prime/nth_prime.py
import math

def is_prime(n):

    if n == 0 or n == 1:
        return False

    for m in range(2, int(math.ceil(math.sqrt(n)) + 1)):
        if n != m and n % m == 0:
            return False

    return True

def next_prime(n):

    m = n + 1

    while True:
        if is_prime(m):
            return m
        else:
            m += 1

    raise ValueError("")

def prime(number):
    if number == 0:
        raise ValueError ("there is no zeroth prime")
    if number < 0:
        raise ValueError ("number must be positive")
    else:
        cnt = 1
        cur = 2
        while cnt < number:
            cur = next_prime(cur)
            cnt += 1

        return cur

